### PR TITLE
Change from pnglite to libpng for PNG reading

### PR DESCRIPTION
**Pull request by heinrich5991**
This is desirable mainly because libpng is maintained and pnglite is
not. pnglite was last updated in 2007 (15 years ago) and probably has a
lot of security vulnerabilities.

libpng is an actively maintained library also used by browsers like
Firefox or Chromium, so it's less likely to contain security
vulnerabilities, also it's more likely to be packaged by Linux
distributions.

This also refuses to load images of types not supported by pnglite,
which allows us to think about backward compatibility while also
introducing libpng.

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by def-** [comment]
> This also refuses to load images of types not supported by pnglite,
which allows us to think about backward compatibility while also
introducing libpng.

I'm wondering if this is the best approach, it means we will have to keep supporting DDNet versions with libpng, but indexed pngs disabled. I see two alternative approaches:
1. Only forbid indexed png in skin db, but not in client
2. Forbid it in client but with a flag that gets set by https://info2.ddnet.tw/info so that we can switch indexed pngs on in old clients too once we want to allow them

File: src/engine/shared/image_loader.cpp [hunk 1]
```diff
@@ -78,6 +79,57 @@ static void LibPNGDeleteReadStruct(png_structp pPNGStruct, png_infop pPNGInfo)
...
+static bool IsUnsupportedByPnglite(png_structp pPNGStruct, png_infop pPNGInfo)
+{
+	int ColorType = png_get_color_type(pPNGStruct, pPNGInfo);
+	int BitDepth = png_get_bit_depth(pPNGStruct, pPNGInfo);
+	int InterlaceType = png_get_interlace_type(pPNGStruct, pPNGInfo);
+	bool Unsupported = false;
+	switch(ColorType)
+	{
+	//case PNG_COLOR_TYPE_GRAY:
+	//case PNG_COLOR_TYPE_RGB:
```
Comment (Robyt3) [inline src/engine/shared/image_loader.cpp:91]: pnglite and the client support loading RGB format though.

https://github.com/ddnet/ddnet/blob/eb0d48a0eb2479ceedabff5af6c3a2de0dbc8579/src/engine/client/graphics_threaded.cpp#L668-L671

Comment (heinrich5991) [reply src/engine/shared/image_loader.cpp]: Accidentally left in because I was testing it.

**Comment by heinrich5991** [comment]
I don't think this is the best approach, merely an improvement over the status quo. I agree that your two solutions are better, I quickly hacked this up because @Learath2 asked for something to be done quickly.

**Pull request merged**